.circleci/config.yml

```diff
@@ -59,6 +59,38 @@ commands:
             pip3 install -r requirements.txt
             python3 main.py "<<parameters.patterns>>"
 jobs:
+  cannon-go-lint-and-test:
+    docker:
+      - image: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:latest
+    resource_class: medium
+    steps:
+      - checkout
+      - check-changed:
+          patterns: cannon,packages/contracts-bedrock/contracts/cannon
+      - run:
+          name: prep Cannon results dir
+          command: mkdir -p /tmp/test-results
+      - run:
+          name: build Cannon example binaries
+          command: make elf  # only compile ELF binaries with Go, we do not have MIPS GCC for creating the debug-dumps.
+          working_directory: cannon/example
+      - run:
+          name: Cannon Go lint
+          command: |
+            golangci-lint run -E goimports,sqlclosecheck,bodyclose,asciicheck,misspell,errorlint --timeout 5m -e "errors.As" -e "errors.Is" ./...
+          working_directory: cannon
+      - run:
+          name: Cannon Go tests
```

Comment:
The test log ends with:
```
=== Failed
=== FAIL: cannon/mipsevm TestHello (unknown)
hello world!--- PASS: TestHello (0.10s)

DONE 137 tests, 2 skipped, 1 failure in 153.150s

CircleCI received exit code 0
```
So the test passed, but after the test failed? Very confused by the logs here, but the tests should definitely work.

Reply:
I think it may be a bug in the gotestsum test result collector, because the `hello world!` is printed on the same line as the pass, and thus the pass might be missed? I'll try add a new-line...

Reply:
```
=== RUN   TestHello
hello world!
--- PASS: TestHello (0.12s)
=== RUN   TestClaim
started!computing 1000 * 3 + 4
claim 3004 is good!
--- PASS: TestClaim (0.23s)
PASS
coverage: 64.6% of statements in github.com/ethereum-optimism/optimism/cannon/...
ok  	github.com/ethereum-optimism/optimism/cannon/mipsevm	151.170s	coverage: 64.6% of statements in github.com/ethereum-optimism/optimism/cannon/...

=== Skipped
=== SKIP: cannon/mipsevm TestEVM/oracle.bin (0.00s)
    evm_test.go:62: oracle test needs to be updated to use syscall pre-image oracle
    --- SKIP: TestEVM/oracle.bin (0.00s)

=== SKIP: cannon/mipsevm TestState/oracle.bin (0.00s)
    state_test.go:35: oracle test needs to be updated to use syscall pre-image oracle
    --- SKIP: TestState/oracle.bin (0.00s)

DONE 137 tests, 2 skipped in 157.614s
```

The newline fixed it!